### DISCUSSION TITLE
Request for Assistance with API Timeout Issue on Vercel Using Supastarter Template

### DISCUSSION THREAD

**Discussion by RainLin10** [vercel/next.js#65777] [Help]
### Summary

'm not sure if you're familiar with the Supastarter template, but I used it to set up a Next.js project. The root directory contains an [apps] folder for frontend code and a [packages] folder for API code. I deployed the project on Vercel via CI/CD. However, I'm encountering a 504 timeout error when calling the API. I understand that the default maximum duration for Pro users is 15 seconds, and despite numerous attempts to change this limit, such as adding a `vercel.json` file with a `maxDuration` setting, nothing seems to work. Could you please advise on how to resolve this issue?

Here's the documentation for Supastarter, which provides more details on the project structure. https://docs.supastarter.dev/nextjs/structure

**Comment by btribouillet** [comment]
Hi @RainLin10 , did you find a solution? I'm facing the exact same issue, I tried to increase the limit on Versel settings but it didn't help.

**Comment by btribouillet** [comment]
Actually increasing the limit on versel settings worked on my side, it was just not instantly deployed.
